neuralprophet/forecaster.py

```diff
@@ -1103,22 +1104,27 @@ def create_df_with_events(self, df, events_df):
                 "The events configs should be added to the NeuralProphet object (add_events fn)"
                 "before creating the data with events features"
             )
-        df_dict, received_unnamed_df = df_utils.prep_copy_df_dict(df)
-        df_dict = self._check_dataframe(df_dict, check_y=True, exogenous=False)
-        if isinstance(events_df, pd.DataFrame):
-            events_df_i = events_df.copy(deep=True)
-        for df_name, df_i in df_dict.items():
-            if isinstance(events_df, dict):
-                events_df_i = events_df[df_name].copy(deep=True)
-            for name in events_df_i["event"].unique():
+        df, received_ID_col, received_single_time_series, received_dict = df_utils.prep_or_copy_df(df)
+        df = self._check_dataframe(df, check_y=True, exogenous=False)
+        df_dict_events = df_utils.create_dict_for_events_or_regressors(df, events_df, "events")
```

Comment:
Here I chose to keep a dict for events and regressors because of the None cases.